src/components/layout/header.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useMemo } from 'react'
+import { Link, useLocation } from 'react-router-dom'
+import { runtime } from 'webextension-polyfill'
+
+import { ArrowBack, Settings } from '../icons'
+import { ROUTES } from '../router-provider'
+
+const Logo = runtime.getURL('assets/images/logo.svg')
+const Close = runtime.getURL('assets/images/close.svg')
+
+const NavigationButton = () => {
+  const location = useLocation()
+
+  const component = useMemo(
+    () =>
+      location.pathname === `/${ROUTES.SETTINGS}` ? (
+        <Link to={ROUTES.INDEX}>
+          <ArrowBack className="h-6" />
+        </Link>
+      ) : (
+        <Link to={ROUTES.SETTINGS}>
+          <Settings className="h-6" />
+        </Link>
+      ),
+
+    [location],
+  )
```

Comment:
useMemo is the one making it to re-render when the location changes. Otherwise it will re-render each time Header component changes its state. I know there is no state at the moment, but I implemented for the long run.